dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -205,11 +210,11 @@ public Set<Tuple> findTransitivelyImpliedCo(Relation co) {
             final MemoryEvent z = (MemoryEvent) t.getSecond();
             final boolean hasIntermediary = mustOut.apply(x).stream().map(Tuple::getSecond)
                     .anyMatch(y -> y != x && y != z &&
-                            (exec.isImplied(x, y) || exec.isImplied(z, y)) &&
+                            (implies(y, x) || implies(y, z)) &&
```

Comment:
The intention is to check `exec(x) => exec(y) || exec(z) => exec(y)`, i.e., that either of the edge events implies some intermediary event `y`. 
The method name "implies" suggests that you try to check the wrong direction though.
However, as I already mentioned, you also inverted the code in "implies" which causes this to be correct again...
In short: you should avoid this double inversion and revert to essentially the original code (you can keep the shorter helper function though).